lib/facter/puppet_settings.rb

```diff
@@ -0,0 +1,13 @@
+require 'puppet'
+
+Facter.add('puppet_ssldir') do
+  setcode do
+    Puppet.settings['ssldir']
```

Comment:
extlib already has a fact for this: https://github.com/voxpupuli/puppet-extlib/blob/master/lib/facter/extlib__puppet_config.rb

Reply:
Neat!  I'll drop that one.